org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/archetype/ArchetypeGenerator.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Konrad Windszus
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Konrad Windszus
+ *******************************************************************************/
+package org.eclipse.m2e.core.ui.archetype;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.m2e.core.project.IArchetype;
+import org.eclipse.m2e.core.project.MavenProjectInfo;
+
+
+/**
+ * Implementations of this OSGi service interface allow to create new Maven projects based on Maven archetypes.
+ * 
+ * @since 2.1.0 (package version 1.0.0)
+ */
+@ProviderType
+public interface ArchetypeGenerator {
+
+  /**
+   * Creates a project structure using the given archetype in non-interactive mode.
+   *
```

Comment:
please have javadoc for params, or you'll get all the javadoc warnings especially since j17+ have turned it up, and for sure it's always a pity to not have these! esp handy is what can be null, what is optional, etc.

Reply:
I remember our integration had to dive into source to understand what "interactive mode" was